tuned/plugins/plugin_irqbalance.py

```diff
@@ -0,0 +1,108 @@
+from . import base
+from .decorators import command_custom
+import tuned.logs
+import errno
+import perf
+
+log = tuned.logs.get()
+
+class IrqbalancePlugin(base.Plugin):
+	"""
+	Plugin for irqbalance settings management.
+	"""
+
+	def __init__(self, *args, **kwargs):
+		super(IrqbalancePlugin, self).__init__(*args, **kwargs)
+		self._cpus = perf.cpu_map()
+		self._irqbalance_sysconfig = "/etc/sysconfig/irqbalance"
+
+	def _instance_init(self, instance):
+		instance._has_dynamic_tuning = False
+		instance._has_static_tuning = True
+
+	def _instance_cleanup(self, instance):
+		pass
+
+	@classmethod
+	def _get_config_options(cls):
+		return {
+			"banned_cpus": None,
+		}
+
+	def _read_irqbalance_sysconfig(self):
+		try:
+			with open(self._irqbalance_sysconfig, "r") as f:
+				return f.read()
+		except IOError as e:
+			if e.errno == errno.ENOENT:
+				log.warn("irqbalance sysconfig file is missing. Is irqbalance installed?")
+			else:
+				log.error("Failed to read irqbalance sysconfig file: %s" % e)
+
+	def _write_irqbalance_sysconfig(self, content):
+		try:
+			with open(self._irqbalance_sysconfig, "w") as f:
+				f.write(content)
+			return True
+		except IOError as e:
+			log.error("Failed to write irqbalance sysconfig file: %s" % e)
+			return False
+
+	def _write_banned_cpus(self, sysconfig, banned_cpumask):
+		return sysconfig + "IRQBALANCE_BANNED_CPUS=%s\n" % banned_cpumask
+
+	def _clear_banned_cpus(self, sysconfig):
+		lines = []
+		for line in sysconfig.split("\n"):
+			if not line.startswith("IRQBALANCE_BANNED_CPUS="):
```

Comment:
It's better to use regex or lstrip. It's enviroment variable and it can begin with whitespaces.